adapters/cpmstar/cpmstar.go

```diff
@@ -0,0 +1,158 @@
+package cpmstar
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+//Adapter :
+type Adapter struct {
+	endpoint string
+}
+
+//MakeRequests :
+func (a *Adapter) MakeRequests(request *openrtb.BidRequest, unused *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var adapterRequests []*adapters.RequestData
+
+	if err := preprocess(request); err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	adapterReq, errors := a.makeRequest(request)
+	if adapterReq != nil {
+		adapterRequests = append(adapterRequests, adapterReq)
+	}
+	errs = append(errs, errors...)
+
+	return adapterRequests, errs
+}
+
+func (a *Adapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
+	var errs []error
+
+	jsonBody, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    jsonBody,
+		Headers: headers,
+	}, errs
+}
+
+func preprocess(request *openrtb.BidRequest) error {
+	for i := 0; i < len(request.Imp); i++ {
+		var imp = request.Imp[i]
+		var bidderExt adapters.ExtImpBidder
+
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			return &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		if err := validateImp(&request.Imp[i]); err != nil {
+			return err
+		}
+
+		var extImp openrtb_ext.ExtImpCpmstar
+		if err := json.Unmarshal(bidderExt.Bidder, &extImp); err != nil {
+			return &errortypes.BadInput{
+				Message: err.Error(),
+			}
+		}
+
+		extImpJSON, err := json.Marshal(extImp)
```

Comment:
This is not needed because the `extImp` is not changed. JSON Marshaling/Unmarshaling is expensive in terms of runtime and memory, so it's best to call them sparingly.